dlt/common/storages/load_package.py

```diff
@@ -514,3 +621,59 @@ def filter_jobs_for_table(
         all_jobs: Iterable[LoadJobInfo], table_name: str
     ) -> Sequence[LoadJobInfo]:
         return [job for job in all_jobs if job.job_file_info.table_name == table_name]
+
+
+@configspec
+class LoadPackageStateInjectableContext(ContainerInjectableContext):
+    storage: PackageStorage
+    load_id: str
+    can_create_default: ClassVar[bool] = False
+    global_affinity: ClassVar[bool] = True
+
+    def commit(self) -> None:
+        with self.state_save_lock:
+            self.storage.save_load_package_state(self.load_id, self.state)
+
+    def on_resolved(self) -> None:
+        self.state_save_lock = threading.Lock()
+        self.state = self.storage.get_load_package_state(self.load_id)
+
+    if TYPE_CHECKING:
+
+        def __init__(self, load_id: str, storage: PackageStorage) -> None: ...
+
+
+def load_package() -> TLoadPackage:
+    """Get full load package state present in current context. Across all threads this will be the same in memory dict."""
+    container = Container()
+    # get injected state if present. injected load package state is typically "managed" so changes will be persisted
+    # if you need to save the load package state during a load, you need to call commit_load_package_state
+    try:
+        state_ctx = container[LoadPackageStateInjectableContext]
+    except ContextDefaultCannotBeCreated:
+        raise Exception("Load package state not available")
```

Comment:
Please derive exception properly. see other similar exceptions.
provide explanation why this is happening. ie when you access this outside of `dlt.destination` decorated function (we inject package only in loader, btw. we can start injecting it in normalize and extract as well)